### PR TITLE
Drop phase.isTyper use in isLegalPrefix/asf

### LINKED ISSUE #17222: Anonymous trait mixin breaks inline match + erasedValue + summonInline
## Compiler version

3.2.2, 3.3.0-RC3

## Minimized code

```Scala
//> using scala "3.3.0-RC3"

import scala.deriving.Mirror
import scala.compiletime.*

trait Reader[-In, Out]

trait A:
  type T
  type F[X]
  type Q = F[T]

object Reader:

  given [X]: Reader[A { type Q = X }, X] with {}    

  type Map2[Tup1 <: Tuple, Tup2 <: Tuple, F[_, _]] <: Tuple = (Tup1, Tup2) match
    case (h1 *: t1, h2 *: t2) => F[h1, h2] *: Map2[t1, t2, F]
    case (EmptyTuple, EmptyTuple) => EmptyTuple    

  inline given productReader[In <: Product, Out <: Product](using mi: Mirror.ProductOf[In])(using mo: Mirror.ProductOf[Out]): Reader[In, Out] = 
    summonAll[Map2[mi.MirroredElemTypes, mo.MirroredElemTypes, Reader]]
    ???

object Test:
  
  trait B[X] extends A:
    type T = X

  trait C extends A:
    type F[X] = X
  
  val bc = new B[Int] with C

  summon[Reader[(bc.type, bc.type), (Int, Int)]]    // fails
```

## Output

```scala
Compiling project (Scala 3.3.0-RC3, JVM)
[error] .\fails.scala:35:49: No given instance of type Reader[(Test.bc : Test.B[Int] & Test.C), Int] was found.
[error] I found:
[error]
[error]     Reader.productReader[In, Out](/* missing */summon[deriving.Mirror.ProductOf[In]]
[error]       )
[error]
[error] But Failed to synthesize an instance of type deriving.Mirror.ProductOf[In]: class Nothing is not a generic product because it is not a case class.
[error]   summon[Reader[(bc.type, bc.type), (Int, Int)]]    // fails
[error]                                                 ^
Error compiling project (Scala 3.3.0-RC3, JVM)
Compilation failed
```

## Expectation

Successful compilation

### PR DESCRIPTION
**Pull request by dwijnand**
Note that "asSeenFrom" (aka asf) is used by SymDenotation#findMember,
which is used by TypeComparer's "hasMatchingMember", as a part of
"compareRefinedSlow".

Previously (using the minimisation in i17222.8.scala) `summonOne` is
inlined during the "inlining" phase, while "summonInline" is inlined
during typing.  The result is that during inlining we fail to find
`Reader[BC, Int]` because we incorrectly consider `A{type F = Int}` as
stable.

Fixes #17222

### CODE REVIEW

File: scaladoc/src/dotty/tools/scaladoc/tasty/ClassLikeSupport.scala [hunk 1]
```diff
@@ -577,7 +577,8 @@ trait ClassLikeSupport:
 
 
   def unwrapMemberInfo(c: ClassDef, symbol: Symbol): MemberInfo =
-    val baseTypeRepr = typeForClass(c).memberType(symbol)
+    val qualTypeRepr = if c.symbol.isClassDef then This(c.symbol).tpe else typeForClass(c)
```
Comment (noti0na1) [inline scaladoc/src/dotty/tools/scaladoc/tasty/ClassLikeSupport.scala:580]: I don't understand this line. Doesn't `This(c.symbol).tpe` return `c.symbol.typeRef`?

Comment (dwijnand) [reply scaladoc/src/dotty/tools/scaladoc/tasty/ClassLikeSupport.scala:580]: No, it returns `c.symbol.thisType` (for which we don't have any more direct syntax in the Quotes API, which is why I went this way).

Comment (noti0na1) [reply scaladoc/src/dotty/tools/scaladoc/tasty/ClassLikeSupport.scala:580]: I see.

**Review by noti0na1 — APPROVED** [review]
LGTM

**Comment by WojciechMazur** [comment]
No longer backportable to 3.6.3 becouse of introduced regression https://github.com/scala/scala3/issues/22070.  Would target 3.6.4 along with the regression fix in https://github.com/scala/scala3/pull/22241

**Pull request merged**